src/main/java/net/rptools/maptool/events/ZoneLoadedListener.java

```diff
@@ -14,28 +14,49 @@
  */
 package net.rptools.maptool.events;
 
-import static net.rptools.maptool.client.functions.MapFunctions.ON_CHANGE_MAP_CALLBACK;
-
 import com.google.common.eventbus.Subscribe;
 import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import net.rptools.maptool.client.MapTool;
 import net.rptools.maptool.client.events.ZoneLoaded;
-import net.rptools.maptool.model.Token;
+import net.rptools.maptool.client.ui.zone.ZoneRenderer;
+import net.rptools.maptool.language.I18N;
+import net.rptools.maptool.model.library.Library;
+import net.rptools.maptool.model.library.LibraryManager;
 import net.rptools.maptool.util.EventMacroUtil;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 
 public class ZoneLoadedListener {
+  private static final Logger LOGGER = LogManager.getLogger(EventMacroUtil.class);
+  public static final String ON_CHANGE_MAP_CALLBACK = "onChangeMap";
 
   public ZoneLoadedListener() {
     new MapToolEventBus().getMainEventBus().register(this);
   }
 
   @Subscribe
   public void OnChangedMap(ZoneLoaded event) {
-    var libTokens = EventMacroUtil.getEventMacroTokens(ON_CHANGE_MAP_CALLBACK);
-    String prefix = ON_CHANGE_MAP_CALLBACK + "@";
-
-    for (Token handler : libTokens) {
-      EventMacroUtil.callEventHandlerOld(
-          prefix + handler.getName(), "", handler, Collections.emptyMap(), true);
+    ZoneRenderer currentZR = MapTool.getFrame().getCurrentZoneRenderer();
+    try {
+      var libs = new LibraryManager().getLegacyEventTargets(ON_CHANGE_MAP_CALLBACK).get();
+      if (!libs.isEmpty()) {
```

Comment:
IMO it's better to exit the method if libs.isEmpty(). Generally it's better to handle the conditions you don't want first. Otherwise they tend to be forgotten. You also get less indent.